### PR TITLE
Permissions

### LINKED ISSUE #77: Readonly mode
Disable edition on global env variable.

### PR DESCRIPTION
**Pull request by decima**
+ Adding global permission management 
+ Fix edit new document
closes #77

### CODE REVIEW

File: assets/js/pages/article/read/ReadArticle.vue [hunk 1]
```diff
@@ -71,36 +71,38 @@
...
-                if(!currentArticle.exists){
+                if (!currentArticle.exists) {
                     this.notFound = true
                 }
             },
-            goToEditArticle(e){
-                if(e.target.tagName !== "INPUT"){
-                    this.$router.push({ path: this.$editLink() });
+            goToEditArticle(e) {
+                if (this.$store.getters.canEdit) {
```
Comment (AlbanCrepel) [inline assets/js/pages/article/read/ReadArticle.vue]: This can be done in a single if statement

File: assets/js/store/modules/Articles.js [hunk 2]
```diff
@@ -28,14 +37,26 @@ const actions = {
...
-            if(e.response.status === 404){
+            if (e.response.status === 404) {
                 commit("setCurrentArticle", null);
                 return null;
             }
         }
     },
+    async loadSettings({commit}) {
+        const item = (await axios.get(Router.url("knowledge_settings"))).data;
+        console.log(item)
```
Comment (AlbanCrepel) [inline assets/js/store/modules/Articles.js]: remove this log

**Review by AlbanCrepel — COMMENTED** [review]
In the PR, we could improve the way to load permissions on the front : 
For now it's loaded in the `created` hook of ArticleActions. Since it's something not dynamic, we could pass the values in the single twig that we loaded for the frontend. That way, we would get the data once, just as needed.

Also, it seems that the `canDelete` action in the store is useless, because the `loadSettings` one seems to do the job.

File: src/Controller/KnowledgeController.php [hunk 3]
```diff
@@ -102,8 +127,12 @@ public function update(File $file, Request $request, FileLoader $fileLoader, Met
      * @Route("/_delete", methods={"GET"},name="_delete_home")
      * @RouteExposed()
      */
-    public function delete(MetadataManager $metadataManager, Request $request, File $file)
+    public function delete(Permissions $permissions, MetadataManager $metadataManager, Request $request, File $file)
     {
+
+        if (!$permissions->isGranted(Permissions::TYPE_DELETE)) {
+            return $this->json("cannot edit", 403);
```
Comment (decima) [inline src/Controller/KnowledgeController.php]:
```suggestion
            return $this->json("cannot delete", 403);
```

**Pull request merged**